Compilation failure on macOS when --no-default-features is passed

Compiling on MacOS, I would not expect to need the default features of 'x11', 'wayland', and 'winpty'. But the if defined statements in `alacritty/src/window.rs` do not take that case into account, so I get the following build failure:

```
error[E0201]: duplicate definitions with name `set_urgent`:
   --> alacritty/src/window.rs:343:5
    |
334 | /     pub fn set_urgent(&self, is_urgent: bool) {
335 | |         if !is_urgent {
336 | |             return;
337 | |         }
338 | |
339 | |         self.window().request_user_attention(RequestUserAttentionType::Critical);
340 | |     }
    | |_____- previous definition of `set_urgent` here
...
343 |       pub fn set_urgent(&self, _is_urgent: bool) {}
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ duplicate definition

error: aborting due to previous error
```

However, if I leave them defined I end up with dead_code warnings on compilation:

```
warning: associated function is never used: `wayland_display`
   --> alacritty/src/window.rs:401:12
    |
401 |     pub fn wayland_display(&self) -> Option<*mut std::ffi::c_void> {
    |            ^^^^^^^^^^^^^^^
    |
    = note: `#[warn(dead_code)]` on by default
```

which is also not ideal.


### System

OS: macOS

--- a/alacritty/src/window.rs
+++ b/alacritty/src/window.rs
@@ -339,7 +339,7 @@ impl Window {
         self.window().request_user_attention(RequestUserAttentionType::Critical);
     }
 
-    #[cfg(any(not(feature = "x11"), windows))]
+    #[cfg(any(windows, not(any(feature = "x11", target_os = "macos"))))]
     pub fn set_urgent(&self, _is_urgent: bool) {}
 
     pub fn set_outer_position(&self, pos: PhysicalPosition<i32>) {
@@ -395,11 +395,6 @@ impl Window {
     #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
     pub fn wayland_display(&self) -> Option<*mut std::ffi::c_void> {
         self.window().wayland_display()
-    }
-
-    #[cfg(any(not(feature = "wayland"), any(target_os = "macos", windows)))]
-    pub fn wayland_display(&self) -> Option<*mut std::ffi::c_void> {
-        None
     }
 
     #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]

--- a/alacritty/src/window.rs
+++ b/alacritty/src/window.rs
@@ -397,6 +397,11 @@ impl Window {
         self.window().wayland_display()
     }
 
+    #[cfg(not(any(feature = "wayland", target_os = "macos", windows)))]
+    pub fn wayland_display(&self) -> Option<*mut std::ffi::c_void> {
+        None
+    }
+
     #[cfg(all(feature = "wayland", not(any(target_os = "macos", windows))))]
     pub fn wayland_surface(&self) -> Option<&Attached<WlSurface>> {
         self.wayland_surface.as_ref()

--- a/alacritty/src/window.rs
+++ b/alacritty/src/window.rs
@@ -4,7 +4,7 @@ use {
     std::sync::atomic::AtomicBool,
     std::sync::Arc,
 
-    glutin::platform::unix::{EventLoopWindowTargetExtUnix, WindowBuilderExtUnix, WindowExtUnix},
+    glutin::platform::unix::{WindowBuilderExtUnix, WindowExtUnix},
     image::ImageFormat,
 };
 
@@ -13,6 +13,7 @@ use {
 use {
     wayland_client::protocol::wl_surface::WlSurface,
     wayland_client::{Attached, EventQueue, Proxy},
+    glutin::platform::unix::EventLoopWindowTargetExtUnix,
 
     alacritty_terminal::config::Colors,
 

